src/main/scala/game/Board1dArray.scala

```diff
@@ -0,0 +1,32 @@
+package game
+
+import cats.implicits._
+
+import scala.collection.immutable.NumericRange
+
+abstract class Board1dArray(m: Short, n: Short) extends BoardT with GameBoard {
+  var _board: Board1d = Array.ofDim[Byte](m * n)
+
+  val mIndices: NumericRange[Short] = NumericRange[Short](0, m, 1)
+  val nIndices: NumericRange[Short] = NumericRange[Short](0, n, 1)
+
+  final protected def generateMoves(): IndexedSeq[Position] = {
+    for {
+      i <- mIndices
+      j <- nIndices
+      if _board(i * m + j) === 0
+    } yield (i, j)
+  }
+
+  protected def board(pos: Position): Byte = {
+    val (i, j) = pos
+    _board(i * m + j)
```

Comment:
having the look up of `i *m` based on ì` index will save the same computation everytime.